File: engines/java/src/main/java/io/github/davecookvectorlabs/windowcleaning/AccessMethod.java
package io.github.davecookvectorlabs.windowcleaning;

/**
 * Access methods for reaching windows on commercial buildings.
 * Each method carries a cost multiplier reflecting equipment, safety,
 * and labour overhead for Northern Ontario conditions.
 */
public enum AccessMethod {
    GROUND_LEVEL(1.0),
    LADDER(1.4),
    BOOM_LIFT(2.2),
    ROPE_ACCESS(3.0),
    SWING_STAGE(3.5);

    private final double multiplier;

    AccessMethod(double multiplier) {
        this.multiplier = multiplier;
    }

    public double getMultiplier() {
        return multiplier;
    }
}

File: engines/php/src/AccessMethod.php
<?php

declare(strict_types=1);

namespace WindowCleaning;

enum AccessMethod: string
{
    case GroundLevel = 'ground_level';
    case Ladder = 'ladder';
    case BoomLift = 'boom_lift';
    case RopeAccess = 'rope_access';
    case SwingStage = 'swing_stage';

    public function multiplier(): float
    {
        return match ($this) {
            self::GroundLevel => 1.0,
            self::Ladder => 1.4,
            self::BoomLift => 2.2,
            self::RopeAccess => 3.0,
            self::SwingStage => 3.5,
        };
    }
}
